### DISCUSSION TITLE
How to get date data from Oracle to time.Time Go with UTC timezone

### DISCUSSION THREAD

**Discussion by thanghv97** [godror/godror#195] [Q&A]
Hi, I have a problem when getting Date data from Oracle to time.Time Go 
I have: 
  - Oracle DB timezone : UTC 
  - My laptop timezone : +07:00

data on DB: 2021-10-17 17:00:00 and when I get from time.Time value 2021-10-17 17:00:00 +0700 +07
I want to get the value is 2021-10-17 17:00:00 +0000 UTC based on the Oracle DB timezone
So is there any solution for that ??

**Comment by tgulacsi** [comment]
It should Just Work.

Give a logger to godror with `godror.SetLogger` (from a logger under github.com/go-kit/log or github.com/UNO-SOFT/ulog) and see what it prints about the server's time zone.

You can force the server's (observed) time zone with the "timezone=" parameter in the connection string.

**Comment by thanghv97** [comment]
Thanks for replying 
I set timezone=UTC in the connection string but I still get a time value +7
![MicrosoftTeams-image (1)](https://user-images.githubusercontent.com/45124560/140692378-eddd6394-d6a2-4b67-a23d-dd1f396e61c9.png)
Log of godror after I set timezone=UTC 
![MicrosoftTeams-image (2)](https://user-images.githubusercontent.com/45124560/140693249-55d8969e-18d5-464e-8700-aa27cb589ffd.png)

**Reply by calebeaires** [reply]
Hey @thanghv97 , did you had success on changing timezone at the connection string?

**Comment by tgulacsi** [comment]
How do you get that (wrong) time out of the DB? How do you print it?
Can you share some code (NOT image)?

If you check out godror somewhere, what does `go test -run=Timezone -v` print?
And what if you set GODROR_TEST_DSN to point to your DB?

For the test DB it should print
```
    z_test.go:2705: sysdate="2021-11-08T07:33:14" currentDate="2021-11-08T08:33:14" dbTZ="+00:00" TZR="+01:00" dbOSTZ="+00:00" ts="2021-11-08T07:33:14.447793Z"    z_test.go:2716: db timezone: +00:00 session timezone: UTC systimestamp: TZR=+00:00 TZD= TZH:TZM=+00:00 localtimestamp: 08-NOV-21 07.33.14.497828 AM
    z_test.go:2720: local: 2021-11-08T08:33:14+01:00
    z_test.go:2727: db: 2021-11-08T07:33:14Z
    z_test.go:2720: local: 2022-05-08T08:33:14+02:00
    z_test.go:2727: db: 2022-05-08T06:33:14Z
```

For my local DB
```
    z_test.go:2705: sysdate="2021-11-08T08:36:51" currentDate="2021-11-08T08:36:51" dbTZ="+02:00" TZR="+01:00" dbOSTZ="+01:00" ts="2021-11-08T08:36:51.958527+01:00"
    z_test.go:2716: db timezone: +02:00 session timezone: UTC systimestamp: TZR=+01:00 TZD= TZH:TZM=+01:00 localtimestamp: 21-NOV-08 07.36.51,983825
    z_test.go:2720: local: 2021-11-08T08:36:51+01:00
    z_test.go:2727: db: 2021-11-08T08:36:51+01:00
    z_test.go:2720: local: 2022-05-08T08:36:51+02:00
    z_test.go:2727: db: 2022-05-08T08:36:51+02:00
```
